feder/institutions/models.py

```diff
@@ -58,6 +61,17 @@ def __str__(self):
     def get_absolute_url(self):
         return reverse("institutions:details", kwargs={"slug": self.slug})
 
+    def get_voivodeship(self):
+        """
+        Returns voivodeship by searching top level parent of related JST.
+        This operation may be expensive, so in queries it should be used with
+        with_voivodeship manager method.
+        """
+        jst = self.jst
+        while jst and jst.parent_id:
+            jst = jst.parent
+        return jst.name
```

Comment:
Could you return just "jst" there to make it reusable in future for other use case?